src/devices/Ssd1331/Ssd1331.cs

```diff
@@ -0,0 +1,1366 @@
+        /// <param name="y2">Ending row</param>
+        /// <param name="col">Line color</param>
+        public void Line(byte x1, byte y1, byte x2, byte y2, Color col)
+        {
+            if (x1 > Width)
+            {
+                x1 = Width;
+            }
+
+            if (y1 > Height)
+            {
+                y1 = Height;
+            }
+
+            if (x2 > Width)
+            {
+                x2 = Width;
+            }
+
+            if (y2 > Height)
+            {
+                y2 = Height;
+            }
+
+            byte lC;
+            byte lB;
+            byte lA;
+
+            (lC, lB, lA) = ToRGB(col);
+            FillEnable(false);
+            SendCommand(Ssd1331DrawLine, x1, y1, x2, y2, lC, lB, lA);
+        }
+
+        /// <summary>
+        /// Set a pixel
+        /// </summary>
+        /// <param name="x">Pixel column</param>
+        /// <param name="y">Pixel row</param>
+        /// <param name="col">Pixel color</param>
+        public void Pixel(byte x, byte y, Color col)
+        {
+            byte low;
+            byte high;
+
+            (high, low) = ToHighLow(col);
+
+            if (x < 0 || x > Width || y < 0 || y > Height)
+            {
+                return;
+            }
+
+            SetColumnAddress(x, x);
+            SetRowAddress(y, y);
+            SendData(high, low);
+        }
+
+        /// <summary>
+        /// Set Contrast for all colors
+        /// </summary>
+        /// <param name="contrast">Contrast level 0-9</param>
+        public void Contrast(int contrast)
+        {
+            int v = contrast * 20;
+
+            if (v > 180)
+            {
+                v = 180;
+            }
+
+            SendCommand(Ssd1331SetContrastA, (byte)v);
+            SendCommand(Ssd1331SetContrastB, (byte)v);
+            SendCommand(Ssd1331SetContrastC, (byte)v);
+        }
+
+        /// <summary>
+        /// Set up window to be acted on
+        /// </summary>
+        /// <param name="x">Top-left column</param>
+        /// <param name="y">Top-left row</param>
+        /// <param name="w">Bottom-right column</param>
+        /// <param name="h">Bottom-right row</param>
+        public void Window(int x, int y, int w, int h)
+        {
+            _x = (byte)x;
+            _y = (byte)y;
+
+            _x1 = (byte)x;
+            _x2 = (byte)(x + w - 1);
+            _y1 = (byte)y;
+            _y2 = (byte)(y + h - 1);
+            SetColumnAddress();
+            SetRowAddress();
+            SetColumnAddress(_x1, _x2);
+            SetRowAddress(_y1, _y2);
+        }
+
+        /// <summary>
+        /// Display a 16bit bitmap
+        /// </summary>
+        /// <param name="x">Starting column</param>
+        /// <param name="y">Starting row</param>
+        /// <param name="Name_BMP">Filename of bitmap</param>
+        /// <returns></returns>
+        public int Bitmap16FS(int x, int y, string Name_BMP)

```

Comment:
please rename the Name_MP to fileNameBmp for example, it's more explicit.
And also, I guess for the name of the function, something like DisplayBitmap16bpp would be more explicit